### ISSUE TITLE
Core service will not start

### ISSUE THREAD

**Issue by normie44** [safing/portmaster#336]
I think this is app is going to be amazing! Very excited. Cheers.

**Pre-Submit Checklist**:
Known issues checked. Found something similar but not the same.  I am running windows 10.

**What happened**:
When I start portmaster I am presented with some UI (a black rectangle) that says 'The portmaster core service is not running' plus a loading animation. There is also a button to 'Start core service' but clicking the button has no effect. After 5 to 10 minutes the app gets a 'not responding' error.

**What did you expect to happen?**:
I was expecting the app would move into a running state.

**How did you reproduce it?**:
Closed the app and restarted it many times. Restarted computer and retried etc.

**Debug Information**:
I don't see any option to capture debug info directly via the app. But I have attached a folder with the logs generated in the windows folder you directed me to.

[Logs.zip](https://github.com/safing/portmaster/files/6621133/Logs.zip)

![Portmaster never starts](https://user-images.githubusercontent.com/85601216/121294126-b496be80-c8a1-11eb-9b35-4b852a8e29dc.png)

**Comment by davegson** [comment]
Hey @normie44, thanks for your report! This is strange and should definitely not be happening.

To keep you updated, @dhaavi will have a proper look at this on Friday. The logs will be helpful, thanks for attaching them.

**Comment by davegson** [comment]
also I just remembered, you could try manually starting the Core Service as described here:
https://docs.safing.io/portmaster/install/windows#portmaster-core-does-not-start-at-boot

**Comment by dhaavi** [comment]
Hey @normie44, thanks for the report. Let's get into it:

When trying to start the Core Service, the Windows service manager replied with:
```
ERROR_SERVICE_REQUEST_TIMEOUT
1053 (0x41D)
The service did not respond to the start or control request in a timely fashion.
```

The UI tried to start the Tray/Notifier, which did not seem to even exist:
```
ENOENT: no such file or directory
```

It seems that the installation failed to some extent. If the Tray/Notifier is missing, then the Core Service might have been corrupted during install.

As a quick way to check if there just was a problem during install, please reinstall the Portmaster.

If that does not change it, we need to gather more data to find the issue. For this, please:

1. Open a `cmd` windows with Administrator privileges.
2. Change to the Portmaster dir with: `cd C:\ProgramData\Safing\Portmaster`.
3. Let it generate the direct execution command with `portmaster-start.exe show core`.
4. Run the returned command to start the Core Service manually.
5. Report back with what happened and generated logs (use https://bin.privacytools.io/).

**Comment by STwersky** [comment]
By the way, I would like to suggest that the label that the core service is not running appears only after the actual real check. I can see that I have a core process running in the background and that filtering works, but always when calling the main interface there is this text, which causes a little psychological discomfort. =)

**Comment by normie44** [comment]
@dhaavi Thank you for the reply!

I am not very technical but I think I followed all the directions.
-Uninstalled & reinstalled. Same issue remains.
-Ran the commands you provided. Received a error. Screenshot attached.
-Tried to double click the .exe file but nothing happened surprisingly. 
-The tray notifier did install. Screenshot attached.
-Logs attached

It might be helpful to know I am using a minimalist version of windows where some of the bloatware is removed. It hasn't given me any trouble before but perhaps it's a issue here, not sure but figured I would mention to be safe.

![portmaster start](https://user-images.githubusercontent.com/85601216/121763038-e69c6080-caed-11eb-8a15-3c508912e05d.png)
![Notifier](https://user-images.githubusercontent.com/85601216/121763045-ea2fe780-caed-11eb-925e-30f788ed2682.png)
[logs.zip](https://github.com/safing/portmaster/files/6641556/logs.zip)

**Comment by davegson** [comment]
> I can see that I have a core process running in the background and that filtering works, but always when calling the main interface there is this text, which causes a little psychological discomfort. =)

I second this! Maybe we should create a new issue (I'd come to it next week)

**Comment by dhaavi** [comment]
> I am not very technical

No problem. We'll guide you through it.

> -Ran the commands you provided. Received a error. Screenshot attached.

You used the Powershell instead of the Cmd.
This is not a problem and works as well too, but you'll need to use `.\portmaster-start.exe show core` instead in step 3.

> -The tray notifier did install. Screenshot attached.

Yes, great!

> It might be helpful to know I am using a minimalist version of windows where some of the bloatware is removed. It hasn't given me any trouble before but perhaps it's a issue here, not sure but figured I would mention to be safe.

Ah, yes. Very good point! I already saw some "The RPC server is unavailable." errors in the logs. I've seen these when the Windows "DNS Client" service wasn't running.
There might be a couple issues when essential Windows services are missing, but I do think that the Portmaster should work anyway.
We'll know what is going on when you run it directly.

Also, because of resource allocation changes, @ppacher will be guiding you through this issue from here on.

**Comment by normie44** [comment]
@dhaavi @ppacher Awesome support, thank you.

After inputting the commands you shared Portmaster started up! However, I think there may be a issue with my VPN. I restarted my computer and ran Portmaster several times, each time it seemed to conflict with my VPN (Mullvad).

I am attaching logs and screenshots. In screenshot one you will see the start of the process and my VPN is working. In screenshot 2 the VPN blocks internet. In screenshot 3 my internet fails completely until I restart my computer. These series of events happened every time.

Perhaps it is necessary to turn off my VPN to start Portmaster? If so, would this share my IP address with you guys?
![1](https://user-images.githubusercontent.com/85601216/122133087-ee5f4c00-cdf0-11eb-8f47-745657867b9b.png)
![2](https://user-images.githubusercontent.com/85601216/122133102-f4552d00-cdf0-11eb-97f0-01c56c4b9102.png)
![3](https://user-images.githubusercontent.com/85601216/122133107-f61ef080-cdf0-11eb-9a25-6e43e825a58d.png)

**Comment by davegson** [comment]
Mullvad requires additional setup since both apps (Mullvad and PM) do things with DNS resulting in conflicts. Check https://github.com/safing/portmaster/issues/313#issuecomment-849715343 for ways to resolve the issue

**Comment by davegson** [comment]
> Perhaps it is necessary to turn off my VPN to start Portmaster? If so, would this share my IP address with you guys?

Also for clarification: yes, you would share your IP with every server you connect to. This would include safing.io when visiting our homepage and updates.safing.io when Portmaster downloads a new version or the most recent malware & tracker lists.

We naturally do not store or collect your IP, but others will. Getting Mullvad to work again would be the best option here ;)

**Comment by normie44** [comment]
Thank you! @davegson 

I followed the steps for Mullvad but no luck. After updating Mullvad with custom DNS I fired up portmaster but got similar errors. Probably something I am screwing up so I am attaching image of my mullvad custom dns and logs.

![Mullvad DNS](https://user-images.githubusercontent.com/85601216/122331343-85a7ca80-cee9-11eb-9c46-2306103a31d6.jpg)
[logs.zip](https://github.com/safing/portmaster/files/6667364/logs.zip)

**Comment by ppacher** [comment]
That seems to be correct.  However, I'm missing the `core` directory in the logs.zip you attached. That makes me think that there's still an issue with the Portmaster. Can you check if, **after a reboot and without the VPN**, the portmaster is actually running? Can you open the UI?

**Comment by GuiltHC** [comment]
Could you help me too? Windows 11, error "The Portmaster Core Service is not running", but the service itself is enabled. Reinstalling and restarting the PC did not help.
